app/api/data-fetching/utils.ts

```diff
@@ -119,3 +120,19 @@ export function withHandleError(handler: RouteHandler): RouteHandler {
     }
   };
 }
+
+export function withUseCase<TResult>(
+  handler: (slug: string, useCase: UseCase) => TResult
+) {
+  return (slug: string, params: { useCase: UseCase }): TResult => {
+    if (!('useCase' in params) || params.useCase in UseCase) {
+      throw new APIRouteError(
+        'Invalid useCase',
+        { slug, route: 'withUseCase' },
+        400
+      );
+    }
```

Comment:
maybe you should log the faulty useCase ?

Reply:
Added !